src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,650 @@
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public final void setCheckPattern(String pattern) {
+        checkPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check violation message pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setMessagePattern(String pattern) {
+        messagePattern = pattern;
+    }
+
+    /**
+     * Setter to specify check ID pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setIdPattern(String pattern) {
+        idPattern = pattern;
+    }
+
+    /**
+     * Setter to specify negative/zero/positive value that defines the number
+     * of lines preceding/at/following the suppressing nearby text. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param format a {@code String} value.
+     */
+    public final void setRowInfluence(String format) {
+        rowInfluence = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+        if (event.getViolation() != null) {
+            final FileText fileText = getFileText(event.getFileName());
+            if (fileText != null) {
+                final List<Suppression> suppressions = getSuppressions(fileText);
+                accepted = getNearestSuppression(suppressions, event) == null;
+            }
+        }
+        return accepted;
+    }
+
+    @Override
+    protected void finishLocalSetup() {
+        // No code by default
+    }
+
+    /**
+     * Returns {@link FileText} instance created based on the given file name.
+     *
+     * @param fileName the name of the file.
+     * @return {@link FileText} instance.
+     * @throws IllegalStateException if the file could not be read.
+     */
+
+    private static FileText getFileText(String fileName) {
+        final File file = new File(fileName);
+        FileText result = null;
+
+        if (!file.isDirectory()) {
+            try {
+                result = new FileText(file, StandardCharsets.UTF_8.name());
+            }
+            catch (IOException ex) {
+                throw new IllegalStateException("Cannot read source file: " + fileName, ex);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns the list of {@link Suppression} instances retrieved from the given {@link FileText}.
+     *
+     * @param fileText {@link FileText} instance.
+     * @return list of {@link Suppression} instances.
+     */
+    private List<Suppression> getSuppressions(FileText fileText) {
+        final List<Suppression> suppressions = new ArrayList<>();
+
+        for (int lineNo = 0; lineNo < fileText.size(); lineNo++) {
```

Comment:
I am slightly confused. We have a row influence value, but we still have to scan the entire file, line by line, to determine if there is a suppression? This will also be doubly intensive since there is no file caching going on.

I recommend looking over what we did for `SuppressWithNearbyCommentFilter`.

Reply:
I am not seeing any Done, so I assume this is still an issue.

Reply:
Did the same as in `SuppressWithNearbyCommentFilter` - store them in an array. Done.